### PR TITLE
APEXCORE-558 Change highlight color to red and implement quit command

### PR DESCRIPTION
**Pull request by sanjaypujare**
@amberarrow could you review and merge pls?

### CODE REVIEW

File: engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java [hunk 1]
```diff
@@ -1532,7 +1537,7 @@ private void printWelcomeMessage()
   private void printHelp(String command, CommandSpec commandSpec, PrintStream os)
   {
     if (consolePresent) {
-      os.print("\033[0;93m");
+      os.print(COLOR_CODE);
       os.print(command);
       os.print("\033[0m");
```
Comment (amberarrow) [inline engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java]: Maybe define and use COLOR_RESET for this literal string to match COLOR_CODE ?

**Comment by vrozov** [comment]
@sanjaypujare Please squash commits.

File: engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java [hunk 2]
```diff
@@ -1547,7 +1553,7 @@ private void printHelp(String command, CommandSpec commandSpec, PrintStream os)
     if (commandSpec.requiredArgs != null) {
       for (Arg arg : commandSpec.requiredArgs) {
         if (consolePresent) {
-          os.print(" \033[3m" + arg + "\033[0m");
```
Comment (davidyan74) [inline engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java:1550]: we might as well define " \033[3m" as well?

**Comment by sandeshh** [comment]
Why not allow the user to specify the color? May be a command line option?

**Comment by sanjaypujare** [comment]
@vrozov  @davidyan74  squashed commits, and addressed review comments. Pls commit

@sandeshh  I think it is overkill to provide a command line option for help output formatting.

File: engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java [hunk 3]
```diff
@@ -641,6 +644,10 @@ void printUsage(String cmd)
         null,
         null,
         "Exit the CLI"));
+    globalCommands.put("quit", new CommandSpec(new ExitCommand(),
```
Comment (vrozov) [inline engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java]: If there is no difference between "exit" and "quit", and both share the same CommandSpec instance, can quit be aliased to exit?

Comment (sanjaypujare) [reply engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java]: Yes, alias is possible but I thought the alias is used by users to create their own aliases.

Comment (davidyan74) [reply engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java]: What's the point of this command? Bash for example does not support "quit".

**Comment by davidyan74** [comment]
I just made a comment on the JIRA https://issues.apache.org/jira/browse/APEXCORE-558
I think putting in a fixed color is a bad thing to do because we should not assume user's terminal's background color

File: engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java (hunk 3 above)
Comment (sanjaypujare) [reply engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java]: Similar to my answer in https://issues.apache.org/jira/browse/APEXCORE-559 , I would say most command line tools I have used support "quit" and I think a lot of users would be surprised to find it missing in the Apex Cli. Bash is a Unix shell and I think they want to limit the reserved words as shell commands.

**Comment by amberarrow** [comment]
I actually like Sandesh's suggestion to make it a configurable option -- not with an escape code which is way too obscure for most users but supporting some fixed set of named colors like red, blue, green and yellow for example. We could then internally map these names to the appropriate escape code.

**Comment by amberarrow** [comment]
Found something interesting with the alias command: it allows you to create an alias for a non-existent command:

apex> alias exit quit
Alias exit created.
apex> quit
Invalid command 'quit'. Type "help" for list of commands
apex> exit
Invalid command 'quit'. Type "help" for list of commands

Once you do that, neither command works !

**Comment by davidyan74** [comment]
@amberarrow  same behavior as bash.

**Comment by sanjaypujare** [comment]
@vrozov @amberarrow  pls review and merge as appropriate

File: engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java [hunk 4]
```diff
@@ -184,10 +184,20 @@
   private String forcePrompt;
   private String kerberosPrincipal;
   private String kerberosKeyTab;
+  private String highlightColor = FORMAT_BOLD;
 
   private static String CONFIG_EXCLUSIVE = "exclusive";
   private static String CONFIG_INCLUSIVE = "inclusive";
 
+  private static final String COLOR_RED = "\033[38;5;196m";
```
Comment (vrozov) [inline engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java]: Please move all static to the start of class definition.

File: engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java [hunk 5]
```diff
@@ -1172,6 +1182,13 @@ public void init() throws IOException
         System.setProperty("socksProxyPort", socks.substring(colon + 1));
       }
     }
+    // set highlight color: check env var first and then config
+    String overrideColor = System.getenv(APEX_HIGHLIGHT_COLOR_ENV_VAR_NAME);
```
Comment (vrozov) [inline engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java]: I'd prefer to do lazy initialization: help is not that frequently used.
```
private String getHighlightColor() {
  if (highlightColor == null) {
    highlightColor = System.getenv(APEX_HIGHLIGHT_COLOR_ENV_VAR_NAME);
    if (StringUtils.isBlank(highlightColor)) {
      highlightColor = conf.get(APEX_HIGHLIGHT_COLOR_PROPERTY_NAME, FORMAT_BOLD);
    }
    highlightColor = color.replace("\\e", "\033");
  }
  return highlightColor;
}
```
This way you may also add unit test that verifies getHighlightColor() behavior.

**Review by vrozov — APPROVED** [review]
Looks good to me.

**Pull request merged**